kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/arbs.kt

```diff
@@ -55,6 +71,19 @@ fun <A, B> Arb<A>.map(f: (A) -> B): Arb<B> = object : Arb<B>() {
  */
 fun <A, B> Arb<A>.flatMap(f: (A) -> Arb<B>): Arb<B> = object : Arb<B>() {
    override fun edgecases(): List<B> = this@flatMap.edgecases().flatMap { f(it).edgecases() }
+
+   override fun edgecases(rs: RandomSource): List<B> =
+      NonEmptyList
```

Comment:
@sksamuel this is the primary machine for bind, where it falls back to taking one random sample accordingly.